src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborWriterTests.Array.cs

```diff
@@ -0,0 +1,143 @@
+            AssertHelper.HexEqual(expectedEncoding, actualEncoding);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public static void WriteArray_DefiniteLengthExceeded_ShouldThrowInvalidOperationException(int definiteLength)
+        {
+            using var writer = new CborWriter();
+            writer.BeginWriteArray(definiteLength);
+            for (int i = 0; i < definiteLength; i++)
+            {
+                writer.WriteInt64(i);
+            }
+
+            Assert.Throws<InvalidOperationException>(() => writer.WriteInt64(0));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public static void WriteArray_DefiniteLengthExceeded_WithNestedData_ShouldThrowInvalidOperationException(int definiteLength)
+        {
+            using var writer = new CborWriter();
+            writer.BeginWriteArray(definiteLength);
+            for (int i = 0; i < definiteLength; i++)
+            {
+                writer.BeginWriteArray(definiteLength: 1);
+                writer.WriteInt64(i);
+                writer.EndWriteArray();
+            }
+
+            Assert.Throws<InvalidOperationException>(() => writer.WriteInt64(0));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public static void EndWriteArray_DefiniteLengthNotMet_ShouldThrowInvalidOperationException(int definiteLength)
+        {
+            using var writer = new CborWriter();
+            writer.BeginWriteArray(definiteLength);
+            for (int i = 1; i < definiteLength; i++)
+            {
+                writer.WriteInt64(i);
+            }
+
+            Assert.Throws<InvalidOperationException>(() => writer.EndWriteArray());
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(10)]
+        public static void EndWriteArray_DefiniteLengthNotMet_WithNestedData_ShouldThrowInvalidOperationException(int definiteLength)
+        {
+            using var writer = new CborWriter();
+            writer.BeginWriteArray(definiteLength);
+            for (int i = 1; i < definiteLength; i++)
+            {
+                writer.BeginWriteArray(definiteLength: 1);
+                writer.WriteInt64(i);
+                writer.EndWriteArray();
+            }
+
+            Assert.Throws<InvalidOperationException>(() => writer.EndWriteArray());
+        }
+
+        [Fact]
+        public static void EndWriteArray_ImbalancedCall_ShouldThrowInvalidOperationException()
+        {
+            using var writer = new CborWriter();
+            Assert.Throws<InvalidOperationException>(() => writer.EndWriteArray());
+        }
+    }
+
+    static class ArrayWriterHelper
+    {
+        public static void WriteArray(this CborWriter writer, params object[] values)
```

Comment:
Similarly, I don't think this should be an extension method.  It provokes "Obviously you thought it was useful, you added it in a test; just make it a method on the type now".

```suggestion
        public static void WriteArray(this CborWriter writer, params object[] values)
```